arches/app/search/components/base_search_view.py

```diff
@@ -133,15 +134,15 @@ def create_query_dict(self, query_dict):
     @abstractmethod
     def handle_search_results_query(
         self,
-        search_query_object: Dict,
-        response_object: Dict,
         search_filter_factory: SearchFilterFactory,
         returnDsl: bool,
-    ) -> Tuple[Dict, Dict]:
+    ) -> Tuple[JSONResponse | JSONErrorResponse | Dict, Dict]:
```

Comment:
I don't think we should continue to promote the use of `response` objects (HTTPResponse, JSONResponse, etc..) in anything other than views, like `search.py`.  Components like this should instead return something more appropriate.  This is really a nod to separations of concerns.

Reply:
Yea that makes sense to me. In my latest change I put in a type hint for `str` since there's a pretty strong use case to return search results from cache (already stored in json strings). Now `search_results()` checks whether it's a string or a dict.